src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/BatchExecutionTest.java

```diff
@@ -139,6 +169,33 @@ public void testExecuteBatch1UseBulkCopyAPI() {
         testExecuteBatch1Internal("BulkCopy");
     }
 
+    private void testBatchUpdateCountWith(int numOfInserts, int errorQueryIndex,
+                                          boolean prepareOnFirstPreparedStatement, String prepareMethod, long[] expectedUpdateCount) throws Exception {
+        try (SQLServerConnection connection = PrepUtil.getConnection(connectionString)) {
+            connection.setEnablePrepareOnFirstPreparedStatementCall(prepareOnFirstPreparedStatement);
+            connection.setPrepareMethod(prepareMethod);
+            try (CallableStatement cstmt = connection.prepareCall(AbstractSQLGenerator.escapeIdentifier(ctstable3Procedure) + " @duration=?, @value=?")) {
+                cstmt.setQueryTimeout(2);
```

Comment:
a bit worried about this hardcoded value whether it might cause intermittent failures depending on where the server is

Reply:
I increased the timeout gaps used in testing. The gap should be large enough that intermittent fails shouldn't happen because of latency.